Desktop/utilities/languagemodel.cpp

```diff
@@ -128,9 +227,52 @@ void LanguageModel::setCurrentLanguage(QString language)
 	if (_currentLanguageCode == LanguageInfo::getLanguageCode(_defaultLocale))	removeTranslators();
 	else																		loadQmFilesForLanguage(_currentLanguageCode);
 
+	refreshAll();
+}
+
+void LanguageModel::setDefaultLocaleFromCurrent()
+{
+	setAlternativeLocaleStatic();
+	
+	QLocale::setDefault(currentLocale());
+	
+	static ColumnUtils::doubleF altFuncToString = [&](double dbl, int precision)
+	{
+		return fq(currentLocale().toString(dbl, 'g', precision));
+	};
+
+	static ColumnUtils::toDoubleF altFuncToDouble = [&](const std::string & str, double & dbl)
+	{
+		bool isDouble = false;
+		dbl = currentLocale().toDouble(tq(str), &isDouble);
+
+		return isDouble;
+	};
+
+	static ColumnUtils::toIntF altFuncToInt = [&](const std::string & str, int & intVal)
+	{
+		bool isInt = false;
+		intVal = currentLocale().toInt(tq(str), &isInt);
+
+		return isInt;
+	};
+	
```

Comment:
At first, I was quite puzzled about this code. A little comment could make it easier to read it. 
```suggestion
	// ColumnUtils is in CommonData library and doesn't access Qt (for instance for QLocale), so instead we use a callback.
```